### PR TITLE
Updating the Getting Started

### PR DESCRIPTION
**Pull request by moxiegirl**
- Clarifying two types of install
- Splitting out release notes
- Splitting out two install methods

Update to #833

Signed-off-by: Mary Anthony <mary@docker.com>

ping @aluzzardi @abronan @vieux

### CODE REVIEW

**Comment by vieux** [comment]
ping @aluzzardi

File: docs/install-manual.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+The easiest way to get started with Swarm is to use the
+[official Docker image](https://registry.hub.docker.com/_/swarm/).
+
+1. Pull the swarm image.
+
+		$ docker pull swarm
+
+1. Create a Swarm cluster using the `docker` command.
+
+    ```bash
```
Comment (thaJeztah) [inline docs/install-manual.md]: (Viewing on GitHub) the back ticks and "bash" are visible here

File: docs/install-manual.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+4. Once the manager is running, check your configuration by running `docker info` as follows:
+
+	```bash
+	docker -H tcp://<manager_ip:manager_port> info
+	```
+
+	For example, if you run the manager locally on your machine:
+
+      $ docker -H tcp://0.0.0.0:2375 info
```
Comment (thaJeztah) [inline docs/install-manual.md]: (Viewing on GitHub) This this block needs to be indented with one more space; it isn't rendered as code-block

File: docs/install-manual.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        └ Reserved Memory: 0 B / 514.5 MiB
+       agent-0: 172.31.40.100:2375
+        └ Containers: 0
+        └ Reserved CPUs: 0 / 1
+        └ Reserved Memory: 0 B / 514.5 MiB
+
+  If you are running a test cluster without TLS enabled, you may get an error.
+  In that case, be sure to unset `DOCKER_TLS_VERIFY` with:
+
+    $ unset DOCKER_TLS_VERIFY
```
Comment (thaJeztah) [inline docs/install-manual.md]: Same problem here; not rendered as code block

File: docs/install-manual.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+Swarm supports TLS authentication between the CLI and Swarm but also between
+Swarm and the Docker nodes. _However_, all the Docker daemon certificates and client
+certificates **must** be signed using the same CA-certificate.
+
+In order to enable TLS for both client and server, the same command line options
+as Docker can be specified:
+
+```bash
+swarm manage --tlsverify --tlscacert=<CACERT> --tlscert=<CERT> --tlskey=<KEY> [...]
```
Comment (thaJeztah) [inline docs/install-manual.md]: I arrived here to check if the rewrite resolves https://github.com/docker/swarm/issues/722

Should this also be a  `docker run ..... swarm manage ...`? Or will that become too complicated here with passing the certificates?

**Comment by moxiegirl** [comment]
@thaJeztah Thanks for the comments.  I'll clean that up.  The team, I think, has some discussion around including the `docker-machine` method --- might change again.

**Comment by thaJeztah** [comment]
@moxiegirl yes, I saw the discussion :) no problem, do with the comments whatever is needed

File: docs/index.md [hunk 5]
```diff
@@ -7,305 +7,46 @@ page_keywords: docker, swarm, clustering
...
-```
+## Understand swarm creation
 
-## TLS
+You can create a swarm on your network by pulling the Docker Swarm image. Then, using Docker, you can configure the swarm manager and all the nodes to run Docker Swarm. This method requires that you:
 
-Swarm supports TLS authentication between the CLI and Swarm but also between
-Swarm and the Docker nodes. _However_, all the Docker daemon certificates and client
-certificates **must** be signed using the same CA-certificate.
+* open identical TCP ports on every node
```
Comment (aluzzardi) [inline docs/index.md]: This is a common misconception (due to errors in our previous doc): You can use any port you like on every node, it doesn't have to be the same.

File: docs/install-manual.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+This section tells you how to create a Docker Swarm on your network to use only for debugging and testing purposes. You can also use this type of installation if you are developing custom applications for Docker Swarm or contributing to it.  
+
+> **Caution**: Only use this set up if your network environment is secured by a firewall or other measures.
+
+## Prerequisites 
+
+You install Docker Swarm on a single system which is known as your Docker Swarm manager. You create the cluster, or swarm, on one or more additional nodes on your network. Each node in your swarm must:
+
+* be accessible by the swarm manager across your network
+* the same TCP port open to listen for the swarm manager  (2375)
```
Comment (aluzzardi) [inline docs/install-manual.md]: Same here. Again, the problem was with our original docs.

`2375` was used as an example (because that's the default docker port).

You can use 1234 if you'd like on node-1 and 4444 on node-2 or whatever.

Maybe this should be a variable such as `node-port`?

`swarm join --addr=node-ip:node-port`?

Comment (aluzzardi) [inline docs/install-manual.md]: And the same is true for `discovery.md`, where we use `2375` as an example - it's very confusing.

Comment (moxiegirl) [inline docs/install-manual.md]: Thanks @aluzzardi I can tweak that.  Is there a way to query a node with swarm and find out which node it is listening on?

Comment (aluzzardi) [inline docs/install-manual.md]: That is the address where Docker is listening on. Docker doesn't export that.

It will show up when running `docker info` against Swarm, although the node should have correctly joined before that works

File: docs/index.md [hunk 7]
```diff
@@ -7,305 +7,46 @@ page_keywords: docker, swarm, clustering
 # Docker Swarm
 
 Docker Swarm is native clustering for Docker. It turns a pool of Docker hosts
-into a single, virtual host.
+into a single, virtual host. Because Docker Swarm serves the standard Docker
+API, any tool that already communicates with a Docker daemon can use Swarm to
+transparently scale to multiple hosts. Supported tools include, but are not  limited to, the following:
```
Comment (aluzzardi) [inline docs/index.md]: How about getting rid of those examples? I'm not even sure they work beside `Docker Compose`. /cc @vieux 

How about:

> transparently scale to multiple hosts, including the Docker CLI and Docker Compose.

File: docs/index.md [hunk 8]
```diff
@@ -7,305 +7,46 @@ page_keywords: docker, swarm, clustering
...
-principle. It ships with a simple scheduling backend out of the box, and as
-initial development settles, an API will develop to enable pluggable backends.
-The goal is to provide a smooth out-of-box experience for simple use cases, and
-allow swapping in more powerful backends, like Mesos, for large scale production
-deployments.
+And of course, the Docker client itself is also supported.
 
-## Release Notes
+Like other Docker projects, Docker Swarm follows the "batteries included but
+removable" principle. It ships with a simple scheduling backend out of the box.
```
Comment (aluzzardi) [inline docs/index.md]: How about:

> It ships with a simple default scheduling backend out of the box.

@vieux ?

I feel *simple* is minimizing the default scheduler.

File: docs/index.md [hunk 9]
```diff
@@ -7,305 +7,46 @@ page_keywords: docker, swarm, clustering
...
+As initial development settles, an API will develop to enable pluggable
+backends. The goal is to provide a smooth out-of-box experience for simple use
```
Comment (aluzzardi) [inline docs/index.md]: > for most use cases

File: docs/index.md [hunk 10]
```diff
@@ -7,305 +7,46 @@ page_keywords: docker, swarm, clustering
...
+cases, and allow swapping in more powerful backends, like Mesos, for large scale
```
Comment (aluzzardi) [inline docs/index.md]: > allow swapping alternative backends

Thoughts? /cc @vieux 

Again, I don't want to minimize the default scheduler.

File: docs/index.md [hunk 11]
```diff
@@ -7,305 +7,46 @@ page_keywords: docker, swarm, clustering
...
-```bash
-$ docker run --rm swarm list token://6856663cdefdec325839a4b7e1de38e8
-172.31.40.100:2375
-172.31.40.101:2375
-172.31.40.102:2375
-```
+## Understand swarm creation
 
-## TLS
+You can create a swarm on your network by pulling the Docker Swarm image. Then, using Docker, you can configure the swarm manager and all the nodes to run Docker Swarm. This method requires that you:
```
Comment (aluzzardi) [inline docs/index.md]: > You can create a swarm on your network by pulling the Docker Swarm image.

`docker pull swarm` doesn't do anything beside actually pulling the image.

It's an optional step, `docker run swarm` will do that under the scenes.

The reason we explicitly mentioned `docker pull swarm` in the first place is to make sure we upgrade to the latest version (`docker run foo` doesn't upgrade `foo` if it has already been `run` before).

We could remove the `pull` step altogether

File: docs/index.md [hunk 12]
```diff
@@ -7,305 +7,46 @@ page_keywords: docker, swarm, clustering
...
-Swarm supports TLS authentication between the CLI and Swarm but also between
-Swarm and the Docker nodes. _However_, all the Docker daemon certificates and client
-certificates **must** be signed using the same CA-certificate.
+* open identical TCP ports on every node
+* install Docker on each node
+* create and manage TLS certificates to secure your swarm
 
-In order to enable TLS for both client and server, the same command line options
-as Docker can be specified:
+This manual method is an advanced method and best suited for experienced administrators or programmers contributing to Docker Swarm. The alternative is to use `docker-machine` to install a swarm.
```
Comment (aluzzardi) [inline docs/index.md]: > experienced administrators or programmers contributing to Docker Swarm

Not sure about that... @vieux ?

File: docs/index.md [hunk 13]
```diff
@@ -7,305 +7,46 @@ page_keywords: docker, swarm, clustering
...
+* create and manage TLS certificates to secure your swarm
 
-In order to enable TLS for both client and server, the same command line options
-as Docker can be specified:
+This manual method is an advanced method and best suited for experienced administrators or programmers contributing to Docker Swarm. The alternative is to use `docker-machine` to install a swarm.  
 
-```bash
-swarm manage --tlsverify --tlscacert=<CACERT> --tlscert=<CERT> --tlskey=<KEY> [...]
-```
+Docker Machine is allows you to quickly install a Docker Swarm on cloud providers or inside your own data center. If you have VirtualBox installed on your local machine, you can quickly build and explore Docker Swarm in your local environment. This method automatically generates a certificate to secure your swarm.
```
Comment (aluzzardi) [inline docs/index.md]: :+1:

File: docs/index.md [hunk 14]
```diff
@@ -7,305 +7,46 @@ page_keywords: docker, swarm, clustering
...
 
-Please refer to the [Docker documentation](https://docs.docker.com/articles/https/)
-for more information on how to set up TLS authentication on Docker and generating
-the certificates.
+Using Docker Machine is the best method for users getting started with Swarm for the first time. To try the recommended method of getting started, see [Get Started with Docker Swarm](install-w-machine.md). 
 
-> **Note**: Swarm certificates must be generated with `extendedKeyUsage = clientAuth,serverAuth`.
+If you are an advanced user or interested in contributing, see [Create a swarm for development](install-manual.md).
```
Comment (aluzzardi) [inline docs/index.md]: Again, not limited to advanced users or contributors - this is likely the most common setup.

Not sure how to phrase that.

@vieux ?

File: docs/install-manual.md [hunk 15]
```diff
@@ -0,0 +1,144 @@
+
+# Create a swarm for development
+
+This section tells you how to create a Docker Swarm on your network to use only for debugging and testing purposes. You can also use this type of installation if you are developing custom applications for Docker Swarm or contributing to it.
```
Comment (aluzzardi) [inline docs/install-manual.md]: Same comment here.

Unless you're toying with Swarm, you'll need to follow the manual process.

**Comment by aluzzardi** [comment]
Added some more comments. Overall looks great!

File: docs/index.md (hunk 13 above)
Comment (thaJeztah) [inline docs/index.md]: Small typo I see; "is allows" -> "allows"

File: docs/install-manual.md (hunk 15 above)
Comment (moxiegirl) [inline docs/install-manual.md]: I haven't worked through the manual process end-to-end yet.  I switched horses to the compose method.  I'll have to update the document for it to vet with me.

File: docs/index.md (hunk 10 above)
Comment (moxiegirl) [inline docs/index.md]: @aluzzardi  The word "simple" here is a good thing.  It doesn't mean small it means "easy-to-use" --- is it a complicated or advanced task to use the scheduler?

Comment (vieux) [inline docs/index.md]: I think we tend to use `batteries included but swappable` instead of  `batteries included but removable` no ?

**Comment by ArmandGrillet** [comment]
This is a great addition to the documentation but a use case is still missing. This new documentation covers these two cases:
- A developer wants to discover Swarm: `Create a Docker Swarm`
- A developer wants to test Swarm: `Create a swarm for development`

But some information are missing: what if a developer wants to use Swarm through a data center? If she is using cloud providers then a link to the Docker Machine doumentationc at the end of `Create a Docker Swarm` would be enough. However, if she wants to deploy a swarm in a private data center securely, there is currently no documentation ([#831](https://github.com/docker/swarm/issues/831) & [#834](https://github.com/docker/swarm/issues/834)).

**Comment by moxiegirl** [comment]
@ArmandGrillet Thank you for the feedback.  Thank you for reminding me, I had planned to add a link at the end of the Getting Started.

The new getting started does use `docker-machine` to deploy a swarm onto Virtualbox.  I haven't  yet tested it, but my understanding is it works essentially the same for a cloud provider/data center as it does for Virtualbox;  you just swap out options for the provider.

**Comment by ArmandGrillet** [comment]
@moxiegirl Indeed, the new tutorial using Docker Machine is straightforward. The problem is the current lack of information considering the use of Docker Swarm without Docker Machine, as it has been presented at the DockerCon:
capture d ecran 2015-05-27 a 18 46 17
No Docker Machine, no cloud providers, just three nodes running Docker + a master to manage them. There is currently no help to make this standard Swarm configuration at a "production" level, i.e. with TLS support.

**Comment by moxiegirl** [comment]
@ArmandGrillet Ah, sorry I thought you wanted to deploy with docker-machine. Yes, you are correct, I was going to work through the manual installation demo next to correct this problem.  Though, of course, I don't have a data center I only have EC2 machines for my testing --- I don't have access to a datacenter internally.

**Comment by ArmandGrillet** [comment]
@moxiegirl OK! The configuration above can be reproduced without a datacenter using VMs (without Docker Machine :wink:). Here is a [Vagrantfile to create three Swarm nodes](https://github.com/ArmandGrillet/rails-swarm/blob/master/Vagrantfile), you can then use your computer as the Swarm master.

File: docs/install-manual.md (hunk 6 above)
Comment (moxiegirl) [inline docs/install-manual.md]: Ok.  Thank you for the answer.  I'll sort it out in the content.

File: docs/index.md (hunk 11 above)
Comment (moxiegirl) [inline docs/index.md]: If the swarm step currently **requires** the pull, we should leave it here for now. We can update it later if you remove the need to do it.

**Comment by moxiegirl** [comment]
ping @thaJeztah @aluzzardi @vieux @abronan Please check updates.  If everyone is cool.  I'll pull this in today. I'm going to do the manual work in another pr so they aren't interdependent.

File: docs/install-w-machine.md [hunk 16]
```diff
@@ -0,0 +1,223 @@
+---
+page_title: Get Started with Docker Swarm
+page_description: Swarm release notes
+page_keywords: docker, swarm, clustering, discovery, release, notes
+---
+
+# Get Started with Docker Swarm
+
+This page introduces you to Docker Swarm by teaching you how to create a swarm
+on your localmachine using Docker Machine and VirtualBox. Once you have a feel
```
Comment (abronan) [inline docs/install-w-machine.md]: localmachine -> local machine

File: docs/install-w-machine.md [hunk 17]
```diff
@@ -0,0 +1,223 @@
...
+Using the instructions appropriate to your system architecture, [install Docker
+Machine](http://docs.docker.com/machine/#installation).
+
+## Create a Docker Swarm
+
+Docker Machine gets hosts ready to run Docker containers. Each node in your
+Docker Swarm must have access to Docker to pull images and run them in
+containers. Docker Machine manages all this provisioning for your swarm.
+
+Before you create a swarm with `docker-machine`, you create a TLS token. Using
```
Comment (abronan) [inline docs/install-w-machine.md]: TLS token -> discovery token

tokens are used for discovery and agent registration, this is not related to TLS or secured communications

File: docs/install-w-machine.md [hunk 18]
```diff
@@ -0,0 +1,223 @@
...
+		INFO[0005] Starting VirtualBox VM...                    
+		INFO[0005] Waiting for VM to start...                   
+		INFO[0050] "local" has been created and is now the active machine. 
+		INFO[0050] To point your Docker client at it, run this in your shell: eval "$(docker-machine env local)" 
+		
+3. Load the `local` machine configuration into your shell.
+
+		$ eval "$(docker-machine env local)"
+		
+4. Generate a TLS token using the Docker Swarm image.
```
Comment (abronan) [inline docs/install-w-machine.md]: same

TLS token -> discovery token

File: docs/release-notes.md [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+
+This page shows the cumulative release notes across all releases of Docker Swarm.
+
+## Version 0.3.0 (June 16, 2015)
+
+For complete information on this release, see the
+[0.3.0 Milestone project page](https://github.com/docker/swarm/wiki/0.3.0-Milestone-Project-Page).
+In addition to bug fixes and refinements, this release adds the following:
+
+- **API Compatibility**: Swarm is now 100% compatible with the Docker API. Everything that runs on top of the Docker API should work seamlessly with a Docker Swarm.
```
Comment (vieux) [inline docs/release-notes.md]: The style is not the same as the 0.2.0 release notes.
Here is `** title **` and below it's `* sentence`

Is it ok ?

File: docs/install-w-machine.md (hunk 17 above)
Comment (moxiegirl) [inline docs/install-w-machine.md]: @abronan I'd like to add something about how the swarm is secured when you use machine.  How is it secured?    

Also, what is the difference between "discovery" and "agent registration" -- I had gotten the impression from a conversation on another PR that they are effectively the same?  Or did you mean "agent discovery" and "agent registration"?

File: docs/release-notes.md (hunk 19 above)
Comment (moxiegirl) [inline docs/release-notes.md]: fixed

File: docs/install-w-machine.md (hunk 18 above)
Comment (moxiegirl) [inline docs/install-w-machine.md]: done

File: docs/install-w-machine.md (hunk 17 above)
Comment (abronan) [inline docs/install-w-machine.md]: Yup they are the same :smile: 

Swarm agents are registering their docker endpoints on the discovery service.

For secured communication, it's `docker-machine` generating certificates/keys for TLS usage for you basically.

**Comment by moxiegirl** [comment]
@vieux @abronan updated.  LGTM at will. :-D

File: docs/install-manual.md (hunk 4 above)
Comment (moxiegirl) [inline docs/install-manual.md]: @thaJeztah  I'm not going to do this one. I have to test the entire procedure and rewrite. That, I am going to do on another PR.  Hope that is cool.

**Comment by aluzzardi** [comment]
LGTM - Huge thanks!

Only one comment: manual doc says to `docker pull` before `docker run`, while machine doc only `docker run`s and explains that it will automatically pull.

Either way is just fine, but we should just stick to one in both docs for the sake of consistency.

**Comment by aluzzardi** [comment]
There are a few comments left by @thaJeztah and then we are good to go.

**Comment by aluzzardi** [comment]
Oh, actually: There's a problem with our shell snippets.

We use ```bash which gets correctly rendered in GitHub. However, not so much on the docs site (it spits the word "bash" into the docs): https://docs.docker.com/swarm/

We should remove just use \`\`\` instead of ```bash

File: docs/index.md (hunk 14 above)
Comment (moxiegirl) [inline docs/index.md]: @aluzzardi  how about experienced network administrator?  This is really about guiding users.  This product is still in BETA so really no one would ever put it into production regardless. So, everyone is "trying it out" and the machine is far simpler.

**Comment by thaJeztah** [comment]
> We should remove just use ``` instead of ```bash

Yes, the ```bash notation is a GitHub feature, it's not standard Markdown.

(Alternatively, indent code-blocks with four spaces in stead of using ```, that's currently the convention the Docker docs)

File: docs/install-manual.md (hunk 3 above)
Comment (moxiegirl) [inline docs/install-manual.md]: Current version shows rendered code:
image

**Comment by moxiegirl** [comment]
ping @vieux @aluzzardi @abronan Hey guys, I'm not going to have time to test and redo the manual for you.  I've incorporated all the comments to date.

**Comment by aluzzardi** [comment]
Awesome, thanks!

LGTM

**Comment by moxiegirl** [comment]
Woot! Thanks guys. If I can a free 4 hours or so, I'll take a stab at the manual page.

**Comment by vieux** [comment]
@moxiegirl thanks! I'll write what I can for the mesos integration and then I'll ping you on it :D
